lib/DxilDia/DxcPixDxilDebugInfo.cpp

```diff
@@ -229,7 +229,24 @@ dxil_debug_info::DxcPixDxilSourceLocations::DxcPixDxilSourceLocations(
         auto* S = llvm::dyn_cast<llvm::DIScope>(DL.getScope());
         while (S != nullptr && !llvm::isa<llvm::DIFile>(S))
         {
-            S = S->getScope().resolve(EmptyMap);
+            if(auto Namespace = llvm::dyn_cast<llvm::DINamespace>(S))
+            {
+                // DINamespace has a getScope member (that hides DIScope's)
+                // that returns a DIScope directly, but if that namespace
+                // is at file-level scope, it will return nullptr.
+                if (auto * ContainingScope = Namespace->getScope())
+                {
```

Comment:
nit: LLVM style guide puts `{` on the line above with the `if`/`while`/`for`/etc statement, and omits braces for single line bodies unless the braces meaningfully add to readability.

(see: https://llvm.org/docs/CodingStandards.html)

Reply:
I think we've had this conversation before :-). I'm keeping the formatting of the surrounding code, with the thought that I should clang-format all of the PIX-related code in one go, but I admit I keep putting that part off. Do you prefer that I follow llvm guidelines just for these changed lines, or keep the current formatting (with the expectation that I'll fix everything later), or clang-format the entire file as part of this checkin?

Reply:
I think at a minimum we should remove the braces around the single-line bodies. clang-format doesn’t generally remove those because the readability criteria is subjective.

Reply:
Roger, wilco